src/DataStructures/Tensor/Expressions/Divide.hpp

```diff
@@ -29,18 +29,26 @@ namespace TensorExpressions {
 /// \tparam T2 the denominator operand expression of the division expression
 /// \tparam Args2 the generic indices of the denominator expression
 template <typename T1, typename T2, typename... Args2>
-struct Divide
-    : public TensorExpression<Divide<T1, T2, Args2...>, typename T1::type,
-                              typename T1::symmetry, typename T1::index_list,
-                              typename T1::args_list> {
-  static_assert(std::is_same<typename T1::type, typename T2::type>::value,
+struct Divide : public TensorExpression<
+                    Divide<T1, T2, Args2...>,
+                    typename std::conditional_t<
+                        std::is_same<typename T1::type, DataVector>::value or
+                            std::is_same<typename T2::type, DataVector>::value,
+                        DataVector, double>,
```

Comment:
This is how the other TEs currently work, so no change request for this PR, but perhaps we'll have to generalize this a bit in the future e.g. to support tensors of `ComplexDataVector`. I haven't used `Tensor` with anything but `DataVector` and `double` so far, but in principle it can support more data types than the two.

Reply:
Agreed. Right now it is assumed that the data type is `double` or `DataVector`

Reply:
ok seems fine for now 👍